### PR TITLE
13058 COLIN API failure - Front and Backend Change validation of the first name field in Coop IA UI and/or Corp Party table

### PR DESCRIPTION
**Pull request by lewischenstudio**
*Issue #:* [/bcgov/entity#13058](https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/bcgov/entity/13058)

*Description of changes:*
- change the max length of "firstName" to 20
- change UI before the schema change

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by argush3** [comment]
@lewischenstudio so create-ui doesn't make use of the shared component you just updated?

**Comment by lewischenstudio** [comment]
> @lewischenstudio so create-ui doesn't make use of the shared component for completing party you just updated?

Some components use it. The CompletingParty component uses the first name rule from create ui.

File: tests/unit/AddEditOrgPerson.spec.ts
Comment (severinbeauvais) [inline tests/unit/AddEditOrgPerson.spec.ts:324]: @argush3 Should we be updating the middle name and last name lengths too?

Comment (argush3) [reply tests/unit/AddEditOrgPerson.spec.ts:324]: we are good on last name as it is 30 in COLIN.  I'm just confirming with @rarmitag if he ever sees middle name.  It could potentially run into the same issue though.

**Comment by lewischenstudio** [comment]
I merged bcrs-shared PR with completing-party: 2.0.4 and tried to update local package.json file with this version. It's not available to install. I think we need to push that project to live before we update the max length in first name rules for UIs.

File: tests/unit/AddEditOrgPerson.spec.ts
Comment (argush3) [reply tests/unit/AddEditOrgPerson.spec.ts:324]: @lewischenstudio lets update the middle name field to 20 in the schema and the UIs as it's possible that we get the same scenario as first name

Comment (lewischenstudio) [reply tests/unit/AddEditOrgPerson.spec.ts:324]: Sure.

**Comment by severinbeauvais** [comment]
> I merged bcrs-shared PR with completing-party: 2.0.4 and tried to update local package.json file with this version. It's not available to install. I think we need to push that project to live before we install the new version for completing party and update the max length in first name rules for UIs. Create UI doesn't use the rules from bcrs-shared, so it's fine for now.

You did not follow the instructions in the readme in bcrs-shared-components.

But I ran steps 14-15 anyways and the new version is not available for import.

File: package.json
Comment (severinbeauvais) [inline package.json:19]: You will need to change this to 2.0.6 once your shared components PR is merged and the component is published.

**Review by severinbeauvais — APPROVED** [review]
Code looks fine.

Do not merge yet. Shared component import needs to be updated again.

File: package.json
Comment (severinbeauvais) [reply package.json:19]: Did you look over the changes from v2.0.0 to v2.0.5 to make sure this project is still compatible with them?

Comment (lewischenstudio) [reply package.json:19]: Yes. `npm i` succeeded.

Comment (severinbeauvais) [reply package.json:19]: Fine, but did you look over the changes to the shared component since it was last used in this project? I see it went from v2.0.0 to v2.0.5, so some things were updated. Do those changes affect this UI?

Comment (lewischenstudio) [reply package.json:19]: The latest CP component before my updates was 2.0.3 in the shared project, but Create UI was using 2.0.0. My updates will be in 2.0.5 (current version) and 2.0.6 (after the next merge). This UI doesn't use the CompletingParty from the shared component project, so changing the CP in the shared component won't affect this UI. We have to keep an eye for the Edit UI though.

Comment (severinbeauvais) [reply package.json:19]: I'm confused. Create UI was using 2.0.0 but this UI doesn't use the shared CompletingParty? Are you saying it should not have been imported previously?

Comment (severinbeauvais) [reply package.json:19]: I found this: https://github.com/bcgov/business-create-ui/blob/00f24266b9387dd9d92a6de709e5238f1befb153/src/views/DissolutionFirm/DissolutionFirm.vue#L221

Comment (severinbeauvais) [reply package.json]: Don't merge this PR without testing this ^^

Comment (lewischenstudio) [reply package.json]: Okay. I'll make the test case.

Comment (severinbeauvais) [reply package.json]: I'd be content with you looking at what changed between before-version and now-version and then checking that the code that uses it is still OK :)

**Review by vysakh-menon-aot — APPROVED** [review]

**Review by argush3 — APPROVED** [review]

**Review by rarmitag — APPROVED** [review]

**Comment by lewischenstudio** [comment]
I was making the test case for the completing party component in the DissolutionFirm.spec.ts. 
I then found out this PR was merged 😆 
I must ran out of time ⌛, and I'll be faster next time 😄
I asked @vysakh-menon-aot to include the test case in his PR. Thank you 👍

**Pull request merged**
